test/compilable/test17502.d

```diff
@@ -0,0 +1,30 @@
+class Foo
+{
+    auto foo()
+    out {}
+    body {}
+
+    auto bar()
+    out { assert (__result > 5); }
```

Comment:
As far as the test suite goes, it is not intended to be an example of best practices. It's sole purpose is to test the compiler - so whatever works is fair game. (In fact, it's often terrible style.)